File: ExpenseTrackerAppAPI/ExpenseTrackerApp/Models/Record.cs
namespace ExpenseTrackerApp.Models
{
    public class Record
    {
        public int Id { get; set; }

        //record type meaning income or expense
        public string Type { get; set; }
        public double Value { get; set; }
        public string Note { get; set; }
        public DateTime DateTime { get; set; }
        public User User { get; set; }
        public Account Account { get; set; }
        public Subcategory Subcategory { get; set; }
    }
}

File: ExpenseTracker-UI/src/app/models/record.ts
import { Account } from '../models/account'
import { Subcategory } from './subcategory';
import { User } from './user';
export interface Record {
    id: number;
    type: string;
    value: number;
    note: string;
    date: Date;
    user: User;
    account: Account;
    subcategory: Subcategory;
}